crates/bifrost/src/loglets/local_loglet/log_store_writer.rs

```diff
@@ -190,6 +208,24 @@ impl LogStoreWriter {
         write_batch.put_cf(data_cf, &key.to_bytes(), data);
     }
 
+    fn trim_log(
+        data_cf: &Arc<BoundColumnFamily>,
+        write_batch: &mut WriteBatch,
+        id: u64,
+        old_trim_point: LogletOffset,
+        new_trim_point: LogletOffset,
+    ) {
+        // the old trim point has already been removed on the previous trim operation
+        let from = RecordKey::new(id, old_trim_point.next());
+        // the upper bound is exclusive for range deletions, therefore we need to increase it
+        let to = RecordKey::new(id, new_trim_point.next());
+
+        trace!("Trim log range: [{from:?}, {to:?})");
+        // We probably need to measure whether range delete is better than single deletes for
+        // multiple trim operations
+        write_batch.delete_range_cf(data_cf, &from.to_bytes(), &to.to_bytes());
```

Comment:
For future reference: Consider evaluating the performance of always deleting from offset_min to trim_point instead of specifying the "from" parameter.